test/xpu_api/utilities/utility/forward/move_if_noexcept.pass.cpp

```diff
@@ -0,0 +1,108 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+#include <oneapi/dpl/utility>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+struct A
+{
+    A(const A&) = delete;
+    A& operator=(const A&) = delete;
+
+    A() = default;
+    A(A&&) = default;
+};
+
+struct AMoveNonNoexcept
+{
+    AMoveNonNoexcept(const AMoveNonNoexcept&) = delete;

```

Comment:
I guess `AMoveNonNoexcept` is not the best name for this structure. It results in rvalue reference in `move_if_noexcept` because the struct is not copy constructible.

Strictly speaking, I see the following important use cases for `move_if_noexcept` testing:
* A type T where move ctor is noexcept and copy ctor is default -> result should be T&&
* A type T where move ctor is noexcept and copy ctor is deleted -> result should be T&&
* A type T where move ctor is not noexcept and copy ctor is default -> result is const T&
* A type T where move ctor is not noexcept and copy ctor is deleted -> result is T&&
* A type T where move ctor is deleted and copy ctor is default -> result is const T&
* A type T where move ctor is deleted and copy ctor is deleted -> result is T&&

I propose to make sure all of them are covered and rename helper structures to reflect the purpose of the test.

Reply:
Done